diff --git a/lib/rubygems/commands/update_command.rb b/lib/rubygems/commands/update_command.rb
index 9974b2b..5705412 100644
--- a/lib/rubygems/commands/update_command.rb
+++ b/lib/rubygems/commands/update_command.rb
@@ -44,6 +44,16 @@ class Gem::Commands::UpdateCommand < Gem::Command
     "#{program_name} GEMNAME [GEMNAME ...]"
   end
 
+  def update_rubygems_command
+    "apt-get install rubygems" <<
+    case RUBY_VERSION
+    when /1\.9/
+      "1.9"
+    when /1\.8/
+      "1.8"
+    end
+  end
+
   def execute
     hig = {}
 
@@ -54,6 +64,8 @@ class Gem::Commands::UpdateCommand < Gem::Command
         fail "No gem names are allowed with the --system option"
       end
 
+      exec update_rubygems_command
+
       rubygems_update = Gem::Specification.new
       rubygems_update.name = 'rubygems-update'
       rubygems_update.version = Gem::Version.new Gem::RubyGemsVersion
